ui/StatusQ/src/singleModelItemData.cpp

```diff
@@ -0,0 +1,254 @@
+#include "StatusQ/SingleModelItemData/singleModelItemData.h"
+#include <QDebug>
+
+SingleModelItemData::SingleModelItemData(QObject *parent)
+    : QObject(parent)
+    , m_item(new QQmlPropertyMap(this))
+{
+}
+
+QQmlPropertyMap* SingleModelItemData::item() const
+{
+    return m_item.data();
+}
+
+QAbstractItemModel* SingleModelItemData::sourceModel() const
+{
+    return m_sourceModel.data();
+}
+
+SingleItemFilter* SingleModelItemData::filter() const
+{
+    return m_filter.data();
+}
+
+bool SingleModelItemData::available() const
+{
+    return m_available;
+}
+
+const QStringList& SingleModelItemData::roles() const
+{
+    return m_roles;
+}
+
+int SingleModelItemData::row() const
+{
+    return m_index.row();
+}
+
+void SingleModelItemData::setSourceModel(QAbstractItemModel* sourceModel)
+{
+    if (m_sourceModel == sourceModel)
+        return;
+
+    if (m_sourceModel)
+    {
+        disconnect(m_sourceModel, nullptr, this, nullptr);
+    }
+    m_sourceModel = sourceModel;
+
+    resetIndex();
+
+    if (!m_sourceModel)
+    {
+        emit sourceModelChanged();
+        return;
+    }
+    
+    connect(m_sourceModel, &QAbstractItemModel::modelReset, this, [this]() {
+        resetIndex();
+    });
+    connect(m_sourceModel, &QAbstractItemModel::rowsInserted, this, [this]() {
+        if (!m_index.isValid())
+        {
+            resetIndex();
+        }
+    });
+    connect(m_sourceModel, &QAbstractItemModel::rowsMoved, this, [this](const QModelIndex &parent, int start, int end, const QModelIndex &destination, int row) {
```

Comment:
vast majority of our c++ keeps the limit of 80 cols, maybe we could stick that that convention, wdyt?

Other minor style thing: we use * and & on the type side in general (differently that in qt's code).

Reply:
We have a `.clang-format` in the root dir and your editor should respect that :) (QtCreator does)